### PR TITLE
19726 most likely fix type changing freezing

### PR DESCRIPTION
**Pull request by 0xF013**
https://community.openproject.org/work_packages/19726

good luck trying to reproduce the original issue locally. I couldn't, but probably fixed it

### CODE REVIEW

**Comment by TeatroIO** [comment]
I've prepared a stage to preview changes. <a href='http://0xf013-fix-19726-type-freezing.opf-openproject-2af22bf12c7dfdad4ddf.ttrcloud.com' target='_blank'>Open stage</a> or <a href='http://0xf013-fix-19726-type-freezing.opf-openproject-2af22bf12c7dfdad4ddf.ttrcloud.com/teatro.log' target='_blank'>view logs</a>.

**Comment by NobodysNightmare** [comment]
Am I missing the part where you are not binding the property to the `fieldController` anymore?

I.e. if you are not using `fieldController.isBusy` anymore, you should also be removing the thing where you say `scope.isBusy = state.isBusy`, right?

**Comment by 0xF013** [comment]
@NobodysNightmare do I use scope.isBusy anywhere?

**Comment by NobodysNightmare** [comment]
I totally made that `scope` up...

You are replacing `fieldController.isBusy` with `fieldController.state.isBusy`. My assumption was that `fieldController.isBusy` is somehow bound to the latter. But that assumption could be wrong altogether.

**Comment by 0xF013** [comment]
@NobodysNightmare I am not using fieldController.isBusy anywhere anymore

**Comment by 0xF013** [comment]
@NobodysNightmare the whole busy state is now common to all inplace editors as it was before but it's no longer implemented by binding the instance isBusy to the shared isBusy, instead the views now use the shared isBusy directly (state is the EditableFieldsState reference in the fieldController)

**Comment by NobodysNightmare** [comment]
Gna... that is what I am talking about, but apparently one of us is stupid... and that is probably me ^^

What I want to say:
- you are not using `fieldController.isBusy` anymore
- you did it before
- my assumption: before `fieldController.isBusy` was magically bound to `fieldController.state.isBusy`
- I can't see you removing the magic binding
- my new assumption: you forgot to remove the magic binding

Now you will probably tell me, that magic doesn't exist and I run away in shame...

**Comment by 0xF013** [comment]
@NobodysNightmare the magic binding was this:

```
    $scope.$watch('fieldController.isBusy', function(isBusy) {
      EditableFieldsState.isBusy = isBusy;
    });
    $scope.$watch(function() {
      return EditableFieldsState.isBusy;
    }, function(isBusy) {
      $scope.fieldController.isBusy = isBusy;
    });
```

File: frontend/app/work_packages/directives/work-package-field-directive.js [hunk 1]
```diff
@@ -50,20 +51,11 @@ module.exports = function(WorkPackageFieldService, EditableFieldsState) {
...
     }
-    
-    $scope.$watch('fieldController.isBusy', function(isBusy) {
-      EditableFieldsState.isBusy = isBusy;
-    });
-    $scope.$watch(function() {
-      return EditableFieldsState.isBusy;
-    }, function(isBusy) {
-      $scope.fieldController.isBusy = isBusy;
-    });
```
Comment (NobodysNightmare) [inline frontend/app/work_packages/directives/work-package-field-directive.js:66]: Ah, here it is... okay... In my defense: Apparently I am not stupid, but simply blind...

Thanks for pointing out ^^

**Comment by NobodysNightmare** [comment]
- Code looks good (after a lot of confusion ^^)
- Could not reproduce initial issue locally
- I can't see that the behaviour got worse

--> LGTM :+1:

**Pull request merged**